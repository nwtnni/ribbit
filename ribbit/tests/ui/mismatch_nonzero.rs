#[ribbit::pack(size = 16)]
#[derive(Clone)]
struct A(u16);

#[ribbit::pack(size = 32)]
#[derive(Clone)]
struct B {
    #[ribbit(size = 16, nonzero)]
    a: A,
    b: u16,
}

fn main() {}
