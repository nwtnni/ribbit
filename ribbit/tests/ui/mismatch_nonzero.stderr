error[E0277]: the trait bound `A: ribbit::NonZero` is not satisfied
 --> tests/ui/mismatch_nonzero.rs:9:8
  |
9 |     a: A,
  |        ^ the trait `ribbit::NonZero` is not implemented for `A`
  |
  = help: the following other types implement trait `ribbit::NonZero`:
            std::num::NonZero<i128>
            std::num::NonZero<i16>
            std::num::NonZero<i32>
            std::num::NonZero<i64>
            std::num::NonZero<i8>
            std::num::NonZero<u128>
            std::num::NonZero<u16>
            std::num::NonZero<u32>
          and $N others
note: required by a bound in `assert_impl_all`
 --> tests/ui/mismatch_nonzero.rs:5:1
  |
5 | #[ribbit::pack(size = 32)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `assert_impl_all`
  = note: this error originates in the macro `::ribbit::private::assert_impl_all` which comes from the expansion of the attribute macro `ribbit::pack` (in Nightly builds, run with -Z macro-backtrace for more info)
